client/src/pages/MyPostsPage.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+  Button,
+  Grid,
+  List,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+  Typography,
+} from '@mui/material';
+import { FC, useEffect, useState } from 'react';
+import { useNavigate } from 'react-router';
+import { PostItem } from '../components/PostItem';
+import { PostsContainer } from '../components/PostsContainer';
+import { useAppDispatch, useAppSelector } from '../store/hooks';
+import { getMyPosts } from '../store/posts/getMyPosts';
+import { POST_CREATE } from '../utils/constants';
+import InboxIcon from '@mui/icons-material/Inbox';
+import DraftsIcon from '@mui/icons-material/Drafts';
+import ArticleIcon from '@mui/icons-material/Article';
+import { IPost } from '../utils/interfaces';
+import { PostState } from '../utils/enums';
+
+export const MyPostsPage: FC = () => {
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const { postsList, isLoading } = useAppSelector((state) => state.posts);
+  const [selectedFilter, setSelectedFilter] = useState<number>(0);
+  const [filteredPosts, setFilteredPosts] = useState<IPost[]>(postsList);
+
+  useEffect(() => {
+    dispatch(getMyPosts());
+  }, [dispatch]);
+
+  useEffect(() => {
+    setFilteredPosts(postsList);
+  }, [postsList]);
+
+  const handleFilterList = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
+    index: number,
+    state: PostState
+  ) => {
+    setSelectedFilter(index);
+    setFilteredPosts([...postsList.filter((post) => post.state === state)]);
+  };
+
+  const handleResetFiltersList = (
+    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
```

Comment:
Why do you accept event argument if you don't use it anywhere?

Reply:
Removed